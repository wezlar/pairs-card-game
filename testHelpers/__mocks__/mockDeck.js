const mockDeck = [
  {
    cardIndex: 0,
    hasMatched: false,
    image: 'images/svg-cards/2_of_hearts.svg',
    isFlipped: false,
    name: '2 of hearts',
  },
  {
    cardIndex: 1,
    hasMatched: false,
    image: 'images/svg-cards/3_of_hearts.svg',
    isFlipped: false,
    name: '3 of hearts',
  },
  {
    cardIndex: 2,
    hasMatched: false,
    image: 'images/svg-cards/4_of_hearts.svg',
    isFlipped: false,
    name: '4 of hearts',
  },
  {
    cardIndex: 0,
    hasMatched: false,
    image: 'images/svg-cards/2_of_hearts.svg',
    isFlipped: false,
    name: '2 of hearts',
  },
  {
    cardIndex: 1,
    hasMatched: false,
    image: 'images/svg-cards/3_of_hearts.svg',
    isFlipped: false,
    name: '3 of hearts',
  },
  {
    cardIndex: 2,
    hasMatched: false,
    image: 'images/svg-cards/4_of_hearts.svg',
    isFlipped: false,
    name: '4 of hearts',
  },
];

export default mockDeck;