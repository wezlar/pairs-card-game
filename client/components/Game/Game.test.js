import React from 'react';
import { shallow } from 'enzyme';
import Component from './index';

import mockDeck from '../../../testHelpers/__mocks__/mockDeck';

const mockProps = {
  game: {
    deck: mockDeck,
  },
};

describe('<Game />', () => {
  test('It doesn\'t blow up' , () => {
    const tree = shallow(<Component {...mockProps} />);
    expect(tree.length).toEqual(1);
  });

  test('Renders correctly', () => {
    const main = shallow(<Component {...mockProps} />);
    expect(main.debug()).toMatchSnapshot();
  });
});