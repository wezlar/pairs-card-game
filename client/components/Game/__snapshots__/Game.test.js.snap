// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> Renders correctly 1`] = `
"<div className=\\"game-wrapper\\">
  <styled.div>
    <Component card={{...}} onClick={[Function: onClick]} />
    <Component card={{...}} onClick={[Function: onClick]} />
    <Component card={{...}} onClick={[Function: onClick]} />
    <Component card={{...}} onClick={[Function: onClick]} />
    <Component card={{...}} onClick={[Function: onClick]} />
    <Component card={{...}} onClick={[Function: onClick]} />
  </styled.div>
</div>"
`;
